src/gui/addnewtorrentdialog.cpp

```diff
@@ -427,12 +427,10 @@ bool AddNewTorrentDialog::loadTorrentFile(const QString &source)
 
 bool AddNewTorrentDialog::loadTorrentImpl()
 {
-    const auto torrentID = BitTorrent::TorrentID::fromInfoHash(m_torrentInfo.infoHash());
-
     // Prevent showing the dialog if download is already present
-    if (BitTorrent::Session::instance()->isKnownTorrent(torrentID))
+    if (BitTorrent::Session::instance()->isKnownTorrent(m_torrentInfo.infoHash()))
```

Comment:
>I think you should still use a local variable to avoid double copying it.
And for magnetUri.infoHash() too.

Done.
>Or maybe you can just check the value of BitTorrent::Session::instance()->findTorrent(m_torrentInfo.infoHash()); and skip this function call?

There is another code path when `isKnownTorrent()` returns `true` but `findTorrent()` returns `nullptr`.